From 3ea0eac09703ea067e29c7460afd72c063a6b19f Mon Sep 17 00:00:00 2001
From: John Noller <jnoller@anaconda.com>
Date: Sun, 20 Jul 2025 14:37:44 -0400
Subject: [PATCH] fix convert_lora_to_gguf.py

Co-authored-by: Charles Bousseau <cbousseau@anaconda.com>

convert_lora_to_gguf.py uses relative, unqualified imports that fail when the tools/scripts
are correctly packaged and namespaced. Additionally, there is no main definition to run the script, 
and the logic is contained in the `if __name__ == '__main__':` block breaking the entry point
definitions. 

This patch can be dropped if/when the upstream repo is refactored to follow python package conventions
and the import statement and `if __name__ == '__main__':` block are updated to follow best practices.
---
 convert_lora_to_gguf.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/convert_lora_to_gguf.py
+++ b/convert_lora_to_gguf.py
@@ -25,7 +25,7 @@ import gguf
 from gguf.constants import GGUFValueType
 
 # reuse model definitions from the conversion/ package
-from conversion import LazyTorchTensor, ModelBase, get_model_class
+from llama_cpp_tools.conversion import LazyTorchTensor, ModelBase, get_model_class
 
 logger = logging.getLogger("lora-to-gguf")
 
@@ -330,7 +330,7 @@ def load_hparams_from_hf(hf_model_id: st
     return config.to_dict(), cache_dir
 
 
-if __name__ == '__main__':
+def main():
     args = parse_args()
     logging.basicConfig(level=logging.DEBUG if args.verbose else logging.INFO)
 
